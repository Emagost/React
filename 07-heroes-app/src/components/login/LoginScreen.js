import React, { useContext } from 'react';
import { useNavigate } from 'react-router-dom';
import { AuthContext } from '../../auth/authContext';
import { types } from '../../types/types';

export const LoginScreen = () => {

    let history = useNavigate();
    const {dispatch} = useContext(AuthContext);

    const handleLogin = () =>{
        const action = {
            type: types.login,
            payload: {name:'Emanuel'}
        }

        dispatch(action)

        const lastPath = localStorage.getItem('lastPath') || '/';
        const lastSearch = localStorage.getItem('lastSearch') || '';

        history(lastPath + lastSearch);

    }

    return (
        <div className="container mt-5">
            <h1>Login</h1>
            <hr/>

            <button
                className="btn btn-primary"
                onClick={ handleLogin }
            >
                Login
            </button>

        </div>
    )
}
