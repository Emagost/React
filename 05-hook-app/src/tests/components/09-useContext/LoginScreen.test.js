import React from 'react';
import {mount} from 'enzyme';
import UserContext from '../../../components/09-useContext/UserContext';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';

describe('test <LoginScreen/>', () => {

    const setUser = jest.fn();
    const wrapper = mount(
        <UserContext.Provider value={{
            setUser
        }}>
            <LoginScreen />
        </UserContext.Provider>
    
    );

    test('snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should use setUser function', () => {
        wrapper.find('button').prop('onClick')();
        
        expect(setUser).toHaveBeenCalledWith({
            id:123,
            name:'Emanuel'
        });
    });
});
