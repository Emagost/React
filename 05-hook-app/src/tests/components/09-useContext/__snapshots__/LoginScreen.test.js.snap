// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <LoginScreen/> snapshot 1`] = `
Array [
  <h1>
    LoginScreen
  </h1>,
  <hr />,
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>,
]
`;
